docs/website/content/v0.3/en/components/apid.md

```diff
@@ -0,0 +1,47 @@
+---
+title: 'apid'
+---
+
+When interacting with Talos, the gRPC api endpoint you will interact with directly is apid.
+Apid acts as the gateway for all component interactions.
+Apid provides a mechanism to route requests to the appropriate destination.
+
+We'll use some examples below to illustrate what apid is doing.
+
+When a user wants to interact with a Talos component via `osctl`, there are two flags that control the interaction with apid.
+The `-e | --endpoints` flag is used to denote which Talos node ( via apid ) should handle the connection.
+Typically this is a public facing server.
+*Note* Typically the endpoints are configured in the Talos config file
+The `-n | --nodes` flag is used to denote which Talos node(s) should respond to the request.
+
+For example, if a user wants to interact with `osd`, a command like `osctl -e cluster.talos.dev memory` may be used.
+
+```bash
+$ osctl -e cluster.talos.dev memory
+NODE                TOTAL   USED   FREE   SHARED   BUFFERS   CACHE   AVAILABLE
+cluster.talos.dev   7938    1768   2390   145      53        3724    6571
+```
+
+In this case, `osctl` is interacting with `apid` running on `cluster.talos.dev` and forwarding the request to the `osd` api.
```

Comment:
Another thing worth pointing out is that only instances of `apid` running control plane nodes can be used as an endpoint.

Reply:
That's not true. Any apid can be used as an endpoint.

Reply:
Then we have a bug 🙂. IIRC that was the original design, but we decided to only have the instances of apid that are in masters do any kind of proxying.

Reply:
I'm not really sure what benefit we'd gain from restricting this, but either way it's not relevant to this pr. We can update this when(if?) it gets fixed.

Reply:
I'd rather not document what is a bug as a feature. Documenting the proper behavior is what we should do, and follow up with a fix. We get a fair amount of traffic to our docs every day, and I'd rather paint a picture that is accurate to the design in this case.

Reply:
I'm gonna put something on the agenda for tomorrow to walk through whether or not we want to proxy only on master nodes.

Reply:
I guess it might be important if that proxying depends more on PKI infra, and we do machine-level certificates for cross-apid communication. Then apid might refuse proxied connections from non-masters.

I agree that technically today we can proxy either way, but at the same time this seems to be overkill in general.

Reply:
Exactly this ^. I think running `apid` in proxy mode has security implications, and we should protect it when we get to RBAC.